pkg/autoupdate/autoupdate.go

```diff
@@ -175,7 +175,10 @@ func (ctrl *Controller) sync(ctx context.Context, key string) error {
 		return nil
 	}
 	up := nextUpdate(clusterversion.Status.AvailableUpdates)
-	clusterversion.Spec.DesiredUpdate = &up
+	clusterversion.Spec.DesiredUpdate = &v1.Update{
+		Version: up.Version,
+		Image:   up.Image,
+	}
```

Comment:
Why can not we just use `up` here as `nextUpdate` returning `v1.Release` now?

Reply:
`Spec.DesiredUpdate` is an `Update`, but `nextUpdate` is a `Release`, because it pulls from `AvailableUpdates`.  We need to convert from `Release` to `Update` somewhere.